### PR TITLE
feat(suite): update the send page

### PR DESCRIPTION
**Pull request by dahaca**
## Description

* update the send page

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/10221

### CODE REVIEW

**Comment by jvaclavik** [comment]
I tested your PR and found some issues:

1) Spacing in case of errors is sometimes weird:
Screenshot 2024-01-08 at 14 47 08

And for example this one is fine:
Screenshot 2024-01-08 at 15 00 25


2) dropdown works weird 

https://github.com/trezor/trezor-suite/assets/858321/8b3ac1c8-442e-48c7-a9f5-246bd22fee73


3) send raw has wrong alignment

Screenshot 2024-01-08 at 14 50 55


4) select for currency doesnt work well

Not sure if it's general problem or not

Screenshot 2024-01-08 at 14 52 31

https://github.com/trezor/trezor-suite/assets/858321/b43c027a-8b4b-41fa-9ba4-b542b0a80ec0


5) Not sure about the border color in input
Screenshot 2024-01-08 at 14 54 06
 

6) Separate items better 
Screenshot 2024-01-08 at 15 02 44

7) Styling in code control

different height on hover:

8) locktime

Screenshot 2024-01-08 at 15 09 52


Now I'm going to check the code.

File: packages/components/src/components/form/Switch/Switch.tsx [hunk 1]
```diff
@@ -9,10 +9,10 @@ import {
     focusStyleTransition,
 } from '../../../utils/utils';
 
-const Wrapper = styled.div<Pick<SwitchProps, 'labelPosition'>>`
+const Wrapper = styled.div<Pick<SwitchProps, 'labelPosition' | 'isSmall'>>`
```
Comment (jvaclavik) [inline packages/components/src/components/form/Switch/Switch.tsx:12]: Offtopic (not part of this PR): `isSmall` is a bit weird naming, I think we should reuse the same pattern like in other places:

```typescript
size: "small" | "normal"
```

What do you think?

File: packages/suite/src/views/wallet/send/components/Outputs/index.tsx [hunk 2]
```diff
@@ -1,108 +1,79 @@
...
                 {outputs.map((output, index) => (
                     <motion.div
+                        layout
                         key={output.id}
-                        {...animation}
-                        onAnimationComplete={onAddAnimationComplete}
+                        initial={
+                            // 3 seems to be the initial amount of renders for this component,
+                            // no entrance animation on the first render
+                            index === 0 || renderCount < 4 || disableAnim
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/send/components/Outputs/index.tsx]: I don't know but counting renders seems a bit hacky to me. Can't we do it somehow better?

File: packages/suite/src/views/wallet/send/index.tsx [hunk 3]
```diff
@@ -56,30 +72,38 @@ const SendLoaded = ({ children, selectedAccount }: SendLoadedProps) => {
 
     const { symbol } = selectedAccount.account;
 
+    console.log(props.sendRaw);
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/send/index.tsx]: You probably forgot to remove this console log.

**Review by jvaclavik — COMMENTED** [review]
Just few small comments. More important are some of these visual stuff. 

Otherwise it looks good and we can merge it soon!

**Comment by jvaclavik** [comment]
One more visual thing: 

Screenshot 2024-01-09 at 14 22 46

**Comment by dahaca** [comment]
> 5\. Not sure about the border color in input

Will handle it in a different PR, I was already on the way to remove the success in put state, since it's not a part of the design system

**Comment by dahaca** [comment]
> One more visual thing:

they are small 🙏 

image

File: packages/suite/src/views/wallet/send/components/Outputs/index.tsx (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/views/wallet/send/components/Outputs/index.tsx]: I am not sure and I've sent a dumb amount of time on making this animation work >_< Form id logic was interfering with `AnimatePresence` and i did not want to touch the form hooks much. Also I cannot just use some ref and turn off the animation on the first render because it does not happen on the first render. If you have ideas – shoot! 🙌

File: packages/components/src/components/form/Switch/Switch.tsx (hunk 1 above)
Comment (dahaca) [reply packages/components/src/components/form/Switch/Switch.tsx:12]: Agreed, also noticed that. I didn't do it then because at the time there was 1 switch in the design system and that was more of a legacy prop.

File: packages/suite/src/views/wallet/send/index.tsx (hunk 3 above)
Comment (dahaca) [reply packages/suite/src/views/wallet/send/index.tsx]: 8a3cca46e9

**Comment by dahaca** [comment]
> 8\. locktime

I have created [an issue](https://github.com/trezor/trezor-suite/issues/10587) concerning removal of all the `success` states everywhere, this PR is already enormous.

**Comment by AdamSchinzel** [comment]
The "=" character is not centered between OP return fields. Also the close button is old design.
E76E1CD9-E909-47DC-AD5B-B42AA50E452E

**Comment by dahaca** [comment]
Posting a bug from Honza here for clarity:

 *  the input label does not react to the `send max` toggle populating the input with a value.

image

File: packages/components/src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -144,6 +143,9 @@ export const Dropdown = forwardRef(
         }, [isToggled, content]);
 
         const setToggled = (isToggled: boolean) => {
+            if (!isToggled) {
+                console.log(isToggled);
```
Comment (jvaclavik) [inline packages/components/src/components/Dropdown/Dropdown.tsx]: Probably missed console log

File: packages/components/src/components/Dropdown/Dropdown.tsx [hunk 5]
```diff
@@ -190,12 +195,13 @@ export const Dropdown = forwardRef(
...
             <MoreIcon
-                isToggled={isToggled}
-                size={24}
+                size="small"
+                variant="tertiary"
                 icon="MORE"
-                color={!isDisabled ? theme.TYPE_DARK_GREY : theme.TYPE_LIGHT_GREY}
-                $isDisabled={isDisabled}
-                {...rest}
+                tabIndex={-1}
```
Comment (jvaclavik) [inline packages/components/src/components/Dropdown/Dropdown.tsx:200]: Are you sure about this? It will not be available with TAB, am I right?

File: packages/components/src/components/form/SelectBar/SelectBar.tsx [hunk 6]
```diff
@@ -169,38 +173,46 @@ export const SelectBar: <V extends ValueTypes>(props: SelectBarProps<V>) => JSX.
     const handleKeyboardNav = (e: KeyboardEvent) => {
         const selectedOptionIndex = options.findIndex(option => option.value === selectedOptionIn);
 
+        let option;
         switch (e.key) {
```
Comment (jvaclavik) [inline packages/components/src/components/form/SelectBar/SelectBar.tsx]: nitpick: I would rather use normal `if` instead of `switch` and maybe ternary in inner condition (but `[option]` could be a bit tricky). 
Anyway we don't use any of the feature that switch offers: multiple options, defaut value, returning values etc.

File: packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/Locktime.tsx [hunk 7]
```diff
@@ -104,55 +90,57 @@ export const Locktime = ({ close }: LocktimeProps) => {
...
+        <Card paddingType="large">
+            <StyledNumberInput
+                control={control}
+                name={inputName}
                 inputState={getInputState(error, inputValue)}
                 defaultValue={inputValue}
-                maxLength={MAX_LENGTH.BTC_LOCKTIME}
+                onChange={handleLocktimeChange}
+                rules={rules}
+                // maxLength={MAX_LENGTH.BTC_LOCKTIME}
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/Locktime.tsx]: Is it intended?

File: packages/suite/src/views/wallet/send/components/Outputs/index.tsx (hunk 2 above)
Comment (jvaclavik) [reply packages/suite/src/views/wallet/send/components/Outputs/index.tsx]: I would expect that the library could offer some handler for it. Don't worry about it, if it works, it's OK. Maybe just please mark it as TODO for the future

**Review by jvaclavik — APPROVED** [review]
What a PR! Well done! 

I commented just few small issues that are not blocking it. From visual perspective it's fine, so I'm approving 👍

**Comment by jvaclavik** [comment]
I'm really looking forward to merge this!

File: packages/components/src/components/Dropdown/Dropdown.tsx (hunk 5 above)
Comment (dahaca) [reply packages/components/src/components/Dropdown/Dropdown.tsx:200]: Since `Dropdown` allows for a custom component to act as the anchor for the dropdown, the keyboard navigation logic is not covered by this `IconButton` but rather the `Container`. `IconButton`, being a `button`, was actually stealing the focus from `Container`, causing the bug you have pointed out in your video from the secon point in your first comment :)

File: packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/Locktime.tsx (hunk 7 above)
Comment (dahaca) [reply packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/Locktime.tsx]: Yes, it is, I thought since the error message [is there](https://github.com/trezor/trezor-suite/blob/aa229eeeda83f939525df5d0c2adb012e22c3a26/packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/Locktime.tsx#L89), there is no point in hard limiting the input + the native `maxLength` logic was counting commas in the NumberInput, so the `MAX_LENGTH.BTC_LOCKTIME` would not represent the length of the actual locktime value, further adding confusion.

But yeah, I forgot to clean up, done! 5e15588e4f

File: packages/components/src/components/form/SelectBar/SelectBar.tsx (hunk 6 above)
Comment (dahaca) [reply packages/components/src/components/form/SelectBar/SelectBar.tsx]: good one! 4aeae0a3b2

File: packages/components/src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (dahaca) [reply packages/components/src/components/Dropdown/Dropdown.tsx]: c2ff67446c

File: packages/suite/src/views/wallet/send/components/Outputs/index.tsx (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/views/wallet/send/components/Outputs/index.tsx]: I believe I found the cause and changed the approach! fb59d7f67a

It's probably [this](https://github.com/trezor/trezor-suite/blob/f4f7bf0dea3c04bf1bd4362b6e6cefcca6304f18/packages/suite/src/hooks/wallet/useSendForm.ts#L303) effect. The default state should be loaded synchronously, not as a hook with a reset. That logic also causes a momentary blink...

**Comment by AdamSchinzel** [comment]
> The "=" character is not centered between OP return fields. Also the close button is old design. E76E1CD9-E909-47DC-AD5B-B42AA50E452E

OP issue still present and also found out that if app is in loading state (fetching account or something else) the send actions are wrongly positioned see screen
4181FBC9-2F3C-459E-90D1-F739FEE8C9C6_4_5005_c

**Review by AdamSchinzel — APPROVED** [review]
lgtm
